homeassistant/components/media_player/samsungtv.py

```diff
@@ -0,0 +1,168 @@
+"""
+homeassistant.components.media_player.denon
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Provides an interface to Samsung TV with a Laninterface.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.samsungtv/
+"""
+import logging
+import socket
+
+from homeassistant.components.media_player import (
+    MediaPlayerDevice, SUPPORT_PAUSE, SUPPORT_VOLUME_STEP,
+    SUPPORT_VOLUME_MUTE, SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_NEXT_TRACK, SUPPORT_TURN_OFF,
+    DOMAIN)
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, STATE_OFF,
+    STATE_ON, STATE_UNKNOWN)
+
+CONF_PORT = "port"
+CONF_TIMEOUT = "timeout"
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['samsungctl==0.5.1']
+
+SUPPORT_SAMSUNGTV = SUPPORT_PAUSE | SUPPORT_VOLUME_STEP | \
+    SUPPORT_VOLUME_MUTE | SUPPORT_PREVIOUS_TRACK | \
+    SUPPORT_NEXT_TRACK | SUPPORT_TURN_OFF
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the Samsung TV platform. """
+
+    if not config.get(CONF_HOST):
```

Comment:
Use the [validate_config helper](https://github.com/balloob/home-assistant/blob/dev/homeassistant/helpers/__init__.py#L9) to do this.